tutorials/llm/llama-3/pruning-distillation/introduction.ipynb

```diff
@@ -0,0 +1,190 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "411e6711-60fc-4488-8aa1-c6463cac8695",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "# Pruning and Distillation of Llama 3.1 model with NeMo Framework"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "03fd1cf4-c67a-4b8d-a5e5-46531be0f991",
+   "metadata": {},
+   "source": [
+    "This demonstration showcases performing pruning and distillation on **Llama 3.1-8B** with the [WikiText-103-v1](https://huggingface.co/datasets/Salesforce/wikitext/viewer/wikitext-103-v1) dataset using NeMo Framework. The [WikiText-103-v1](https://huggingface.co/datasets/Salesforce/wikitext/viewer/wikitext-103-v1) language modeling dataset is a collection of over 100 million tokens extracted from the set of verified 'Good' and 'Featured' articles on Wikipedia. \n",
+    "\n",
+    "For this demonstration, we will perform a light finetuning procedure on the `Meta Llama 3.1 8B` teacher model to generate a finetuned teacher model. This finetuned teacher model will then be trimmed. There are two methods to prune a model: depth-pruning and width-pruning. This workflow will showcase both methods which will yield `4b_depth_pruned_model.nemo` and `4b_width_pruned_model.nemo` respectively, that will serve as a starting point for distillation to the final 4B models. \n",
+    "\n",
+    "> We are using models utilizing the `meta-llama/Meta-Llama-3.1-8B` tokenizer for this demonstration.\n",
+    "\n",
+    "> `NOTE:` Ensure that you run this notebook inside the [NeMo Framework container](https://catalog.ngc.nvidia.com/orgs/nvidia/containers/nemo) which has all the required dependencies. \n",
+    "\n",
+    "**Instructions are available in the associated tutorial README to download the model and the container.**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5a5026ce-39f1-43e3-93af-4c4f1e9da1f2",
+   "metadata": {
+    "scrolled": true,
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "!pip install --upgrade ipywidgets notebook\n",
+    "!pip install datasets"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "afe59b07-bb48-4913-90cc-bb416b48196c",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "---\n",
+    "## Prerequisites\n",
+    "Ensure you have the following -\n",
+    "1. **Get the teacher model**: Download the `Meta Llama 3.1 8B .nemo` model. You must follow the instructions in the associated README to download and mount the folder to the NeMo FW container."
```

Comment:
Use full NeMo Framework name

    "1. **Get the teacher model**: Download the `Meta Llama 3.1 8B .nemo` model. You must follow the instructions in the associated README to download and mount the folder to the NeMo Framework container."